File: src/main/java/org/dedda/bratwurst/lang/ClassRegistration.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.lang.scope.Scope;

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
public class ClassRegistration extends BWInstruction {

    private final BWClass bwClass;

    public ClassRegistration(BWClass bwClass) {
        super(0);
        this.bwClass = bwClass;
    }

    @Override
    public void run(Scope scope) {
        scope.registerClass(bwClass);
    }
}

File: src/main/java/org/dedda/bratwurst/lang/ClassRegistration.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.classes.BWClass
import org.dedda.bratwurst.lang.scope.Scope

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
class ClassRegistration(private val bwClass: BWClass) : BWInstruction(0) {
    override fun run(scope: Scope) {
        scope.registerClass(bwClass)
    }

}